Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -1878,3 +1931,352 @@ def parse_date_string(date_string, date_format='%Y-%m-%dT%H:%M:%S'):
+
+    :type indicator: ``str``
+    :param indicator: indicator field. if using file hashes, can be dict
+
+    :type indicator_type: ``str``
+    :param indicator_type:
+        type of indicator ('url, 'domain', 'ip', 'cve', 'email', 'md5', 'sha1', 'sha256', 'crc32', 'sha512', 'ctph')
+
+    :type vendor: ``str``
+    :param vendor: Integration ID
+
+    :type score: ``int``
+    :param score: DBot score (0-3)
+
+    :type description: ``str`` or ``None``
+    :param description: description (will be added to malicious if dbot_score is 3). can be None
+
+    :type build_malicious: ``bool``
+    :param build_malicious: if True, will add a malicious entry
+
+    :return: dbot entry
+    :rtype: ``dict``
+    """
+    if not 0 <= score <= 3:
+        raise DemistoException('illegal DBot score, expected 0-3, got `{}`'.format(score))
+    indicator_type_lower = indicator_type.lower()
+    if indicator_type_lower not in INDICATOR_TYPES:
+        raise DemistoException('illegal indicator type, expected one of {}, got `{}`'.format(
+            INDICATOR_TYPES.keys(), indicator_type_lower
+        ))
+    dbot_entry = {
+        outputPaths['dbotscore']: {
+            'Indicator': indicator,
+            'Type': indicator_type_lower,
+            'Vendor': vendor,
+            'Score': score
+        }}
+    if score == 3 and build_malicious:
+        dbot_entry.update(build_malicious_dbot_entry(indicator, indicator_type, vendor, description))
+    return dbot_entry
+
+
+def build_malicious_dbot_entry(indicator, indicator_type, vendor, description=None):
+    """ Build Malicious dbot entry
+    Examples:
+        >>> build_malicious_dbot_entry('8.8.8.8', 'ip', 'Vendor', 'Google DNS')
+        {'IP(val.Address && val.Address == obj.Address)': {'Malicious': {'Vendor': 'Vendor', 'Description': 'Google DNS\
+'}, 'Address': '8.8.8.8'}}
+
+        >>> build_malicious_dbot_entry('md5hash', 'MD5', 'Vendor', 'Malicious File')
+        {'File(val.MD5 && val.MD5 == obj.MD5 || val.SHA1 && val.SHA1 == obj.SHA1 || val.SHA256 && val.SHA256 == obj.SHA\
+256 || val.SHA512 && val.SHA512 == obj.SHA512 || val.CRC32 && val.CRC32 == obj.CRC32 || val.CTPH && val.CTPH == obj.CTP\
+H || val.SSDeep && val.SSDeep == obj.SSDeep)': {'Malicious': {'Vendor': 'Vendor', 'Description': 'Malicious File'}\
+, 'MD5': 'md5hash'}}
+
+    :type indicator: ``str``
+    :param indicator: Value (e.g. 8.8.8.8)
+
+    :type indicator_type: ``str``
+    :param indicator_type: e.g. 'IP'
+
+    :type vendor: ``str``
+    :param vendor: Integration ID
+
+    :type description: ``str``
+    :param description: Why it's malicious
+
+    :return: A malicious DBot entry
+    :rtype: ``dict``
+    """
+    indicator_type_lower = indicator_type.lower()
+    if indicator_type_lower in INDICATOR_TYPES:
+        key = INDICATOR_TYPES[indicator_type_lower]
+        # `file` indicator works a little different
+        if key == 'file':
+            key = indicator_type_lower.upper()
```

Comment:
lower upper upper lower, too complex, make the code longer if you need to, just not confusing.
If more lines make it more readable then add those lines